components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/password/PasswordRecoveryManagerImpl.java

```diff
@@ -202,6 +207,75 @@ public PasswordResetCodeDTO confirm(String confirmationCode, String tenantDomain
         return buildPasswordResetCodeDTO(confirmationCode);
     }
 
+    /**
+     * Validate the code given for password recovery and return the password reset code.
+     *
+     * @param otp              One Time Password.
+     * @param confirmationCode Confirmation code.
+     * @param tenantDomain     Tenant domain.
+     * @param properties       Meta properties in the confirmation request.
+     * @return PasswordResetCodeDTO {@link PasswordResetCodeDTO} object which contains password reset code.
+     * @throws IdentityRecoveryException Error while confirming password recovery.
+     */
+    @Override
+    public PasswordResetCodeDTO confirm(String otp, String confirmationCode, String tenantDomain,
+                                        Map<String, String> properties) throws IdentityRecoveryException {
+
+        validateTenantDomain(tenantDomain);
+        UserAccountRecoveryManager userAccountRecoveryManager = UserAccountRecoveryManager.getInstance();
+        // In the recovery scenarios which are not OTP based, the confirmation code is a combination of the
+        // recovery flow id and another UUID generated. Hence, we need to get the recovery flow id from the
+        // confirmation code. In the OTP based recovery scenario, the confirmation code is the recovery flow id.
```

Comment:
shall we make this a multi-line comment

Reply:
Fixed with 4c8c9f64b52364e581f4f4f1890afe0f11577c08